src/modeler/AutoRest.Swagger/SwaggerModeler.cs

```diff
@@ -179,7 +183,11 @@ public override IEnumerable<LogMessage> Compare()
 
             // Look for semantic errors and warnings in the new document.
             var validator = new RecursiveObjectValidator(PropertyNameResolver.JsonName);
-            var LogMessages = validator.GetValidationExceptions(newDefintion.FilePath, newDefintion).ToList();
+            var LogMessages = validator.GetValidationExceptions(newDefintion.FilePath, newDefintion, new ServiceDefinitionMetadata
```

Comment:
Not legacy, just redundant code lying around scheduled to be cleaned up soon. Had to add defaults to appease the compiler. @olydis can you confirm this